ocaml/typing/typedecl.ml

```diff
@@ -782,7 +782,8 @@ let transl_declaration env sdecl (id, uid) =
       let cty = transl_simple_type ~new_var_jkind:Any env ~closed:no_row Mode.Alloc.Const.legacy sty in
       Some cty, Some cty.ctyp_type
   in
-  let any = Jkind.Primitive.any ~why:Initial_typedecl_env in
+  let any = Jkind.Type.Primitive.any ~why:Initial_typedecl_env in
+  let any' = Jkind.Primitive.any ~why:Initial_typedecl_env in
```

Comment:
What's going on here? This seems fishy. If it's necessary, please add an explanatory comment.

Reply:
As far as I see, the existing `any` was just used as a short-hand, and the alternative would be to add a promotion at each use site. This is related to whether there should be `Jkind.Primitive`s in the first place or just `Jkind.Type.Primitive`s.

Reply:
What do you mean by "add a promotion"? Oh... I think you mean a conversion to `Jkind.t`. I think we'll need both `any` (which will be the top of the type-kind lattice) and `top` (which will be the top of the kind lattice).

Reply:
Yes, we also agreed with Leo `top` would be necessary (from my perspective it seemed the compiler creates too many type variables with the `any` kind where it should be `top`, or generally just assumes things are `any` rather than `top`).

Reply:
I made an attempt at moving to `top` where appropriate. There are many such places, especially in `ctype`, where this seems necessary.